### ISSUE TITLE
Configure scope in async method do nothing

### ISSUE THREAD

**Issue by ketourneau** [getsentry/sentry-unity#224]
### Environment

How do you use Sentry?
Sentry SaaS

Which SDK and version?
sentry-unity 0.1.1
Unity 2020.3.11f1 LTS (iOS platform)
MacOS Big Sur 11.4

### Steps to Reproduce

1. Create new empty unity project
2. Add sentry-unity 0.1.1
3. Call SentryUnity.Init and SentrySdk.ConfigureScope (with userId) in async Start method
```csharp
async void Start() {
    SentryUnity.Init((options) => {
        options.Dsn = tmpdDsn;
        options.Environment = "development";
        options.Release = "1.2.3";
        options.CaptureInEditor = true;
        options.Enabled = true;
    });
    
    SentrySdk.ConfigureScope(scope => {
        scope.User = new Sentry.User() { Id = "AAA" };
    });
}
```
4. Log Error
```csharp
[ContextMenu("Test")]
void Test() {
    Debug.LogError("Test");
}
```

### Expected Result

We expect to see userId in sentry.io.

### Actual Result

We can't see userId in sentry.io.
If we remove `async` before start it works (but we need to configure scope from async method in our app).

**Comment by bruno-garcia** [comment]
Currently the scope is bound to an async flow, that's why the behavior you're seeing.
We have an issue tracking this scenario here: https://github.com/getsentry/sentry-dotnet/issues/628

To help validate our understanding, do you have the expectation that any scope change (any call to ConfigureScope) should affect any event captured? So any `Debug.LogException` call in any thread/part of the game should include all breadcrumbs, tags, the user info, etc. Set by any other thread?

**Comment by Blocovin** [comment]
Thank you for your response.

Yes we expect any call on ConfigureScope (from another thread or not) affect any event captured.

We found a workaround by instantiate a GameObject (from async method) that call ConfigureScope from Start method.

**Comment by bruno-garcia** [comment]
Resolves in #260 

It'll be included on version `0.4.2`

It makes it so that any call to `SentrySdk.SetTag` or `AddBreadcrumb` or anything else that you do through `SentrySdk.ConfigureScope` mutates a single, static `Scope` object that affects **all threads of the app**. This is useful for Desktop and Mobile apps when you have a single user session and you add context to Sentry anywhere in the app and want that to be included if it crashes in any other thread.
